stdlib/enum.pyi

```diff
@@ -193,14 +193,12 @@ class IntEnum(int, _IntEnumBase):
 
 def unique(enumeration: _EnumerationT) -> _EnumerationT: ...
 
-_auto_null: Any
-
-# subclassing IntFlag so it picks up all implemented base functions, best modeling behavior of enum.auto()
-class auto(IntFlag):
-    _value_: Any
-    @_magic_enum_attr
-    def value(self) -> Any: ...
-    def __new__(cls: type[Self]) -> Self: ...
+class auto:
+    value: Never
+    # This is a lie, auto() returns an auto instance. But auto instances
+    # are solely intended to be used inside the body of Enum sub-classes
+    # and get replaced with Enum instances at runtime.
+    def __new__(cls: type[Self]) -> Enum: ...  # type: ignore[misc]
```

Comment:
There's no need to annotate `cls` in a `__new__` method if there isn't a type-dependency between the type of `cls` and the return type.

```suggestion
    def __new__(cls) -> Enum: ...
```